src/scanner.jl

```diff
@@ -140,25 +140,98 @@ function get_mark(stream::TokenStream)
     Mark(stream.index, stream.line, stream.column)
 end
 
+# -------------
+# forwardchars!
+# -------------
 
-# Advance the stream by k characters.
-function forwardchars!(stream::TokenStream, k::Integer=1)
-    for _ in 1:k
+# Advance the stream by a chacater and the index.
+function forwardchar_skip!(stream::TokenStream)
+    forward!(stream.input)
+    stream.index += 1
+    nothing
+end
+
+# Advance the stream by a chacater and the index and a column.
+function forwardchar_nobreak!(stream::TokenStream)
+    forward!(stream.input)
+    stream.index += 1
+    stream.column += 1
+    nothing
+end
+
+# Advance the stream by a chacater and the index and break a line.
+function forwardchar_breakline!(stream::TokenStream)
+    forward!(stream.input)
+    stream.index += 1
+    stream.column = 0
+    stream.line += 1
+    nothing
+end
+
+# YAML 1.1 [22] b-line-feed ::= #xA /*LF*/
+# YAML 1.2 [24] b-line-feed ::= x0A
+const b_line_feed = '\n'
+
+# YAML 1.1 [23] b-carriage-return ::= #xD /*CR*/
+# YAML 1.2 [25] b-carriage-return ::= x0D
+const b_carriage_return = '\r'
+
+# YAML 1.1 [24] b-next-line ::= #x85 /*NEL*/
+const yaml_1_1_b_next_line = '\u85'
+
+# YAML 1.1 [25] b-line-separator ::= #x2028 /*LS*/
+const yaml_1_1_b_line_separator = '\u2028'
+
+# YAML 1.1 [26] b-paragraph-separator ::= #x2029 /*PS*/
+const yaml_1_1_b_paragraph_separator = '\u2029'
+
+# Advance the stream by `n` characters.
+# YAML 1.1 [28] b-specific ::= b-line-separator | b-paragraph-separator
+yaml_1_1_is_b_specific(c::Char) = c == yaml_1_1_b_line_separator || c == yaml_1_1_b_paragraph_separator
+# YAML 1.1 [29] b-generic ::= ( b-carriage-return b-line-feed) | b-carriage-return | b-line-feed | b-next-line
+# YAML 1.1 [33] b-ignored-any ::= b-generic | b-specific
+function forwardchars!(::YAMLVersion{Symbol("1.1")}, stream::TokenStream, n::Integer=1)
+    i = 1
+    while i ≤ n
         c = peek(stream.input)
-        forward!(stream.input)
-        stream.index += 1
-        if in(c, "\n\u0085\u2028\u2029") ||
-            (c == '\r' && peek(stream.input) == '\n')
-            stream.column = 0
-            stream.line += 1
+        if c == b_carriage_return
+            forwardchar_breakline!(stream)
+            i += 1
+            if peek(stream.input) == b_line_feed
+                forwardchar_skip!(stream)
+                i += 1
+            end
+        elseif c == b_line_feed || c == yaml_1_1_b_next_line || yaml_1_1_is_b_specific(c)
+            forwardchar_breakline!(stream)
+            i += 1
         else
-            stream.column += 1
+            forwardchar_nobreak!(stream)
+            i += 1
         end
     end
-    stream.index += k
-    nothing
 end
 
+# Advance the stream by `n` characters.
+# YAML 1.2 [28] b-break ::= ( b-carriage-return b-line-feed ) | b-carriage-return | b-line-feed
+function forwardchars!(::YAMLVersion{Symbol("1.2")}, stream::TokenStream, n::Integer=1)
```

Comment:
It would be nicer if we could have just one `forwardchars!` function for both versions and dispatch on the version where they do differ.